File: 351004/Beliaev/restapp/src/main/java/com/example/restapp/service/StickerService.java
package com.example.restapp.service;

import com.example.restapp.dto.request.StickerRequestTo;
import com.example.restapp.dto.response.StickerResponseTo;
import com.example.restapp.exception.EntityNotFoundException;
import com.example.restapp.mapper.StickerMapper;
import com.example.restapp.model.Sticker;
import com.example.restapp.repository.StickerRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class StickerService {
    private final StickerRepository repository;
    private final StickerMapper mapper;

    public StickerResponseTo create(StickerRequestTo request) {
        Sticker sticker = mapper.toEntity(request);
        Sticker saved = repository.save(sticker);
        return mapper.toResponse(saved);
    }

    public List<StickerResponseTo> getAll() {
        return repository.findAll().stream()
                .map(mapper::toResponse)
                .collect(Collectors.toList());
    }

    public StickerResponseTo getById(Long id) {
        return repository.findById(id)
                .map(mapper::toResponse)
                .orElseThrow(() -> new EntityNotFoundException("Sticker not found with id: " + id));
    }

    public StickerResponseTo update(Long id, StickerRequestTo request) {
        Sticker sticker = repository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Sticker not found with id: " + id));
        mapper.updateEntityFromDto(request, sticker);
        repository.update(sticker);
        return mapper.toResponse(sticker);
    }

    public void delete(Long id) {
        if (!repository.deleteById(id)) {
            throw new EntityNotFoundException("Sticker not found with id: " + id);
        }
    }
}

File: 351004/Barsukov/app/services/sticker_service.py
from app.repository.memory import sticker_repo
from app.schemas.sticker import StickerRequestTo, StickerResponseTo
from app.core.exceptions import AppException


class StickerService:
    def create(self, dto: StickerRequestTo) -> StickerResponseTo:
        return StickerResponseTo(**sticker_repo.save(dto.model_dump()))

    def get_all(self):
        return [StickerResponseTo(**s) for s in sticker_repo.find_all()]

    def get_by_id(self, id: int):
        res = sticker_repo.find_by_id(id)
        if not res:
            raise AppException(404, "Sticker not found", 9)
        return StickerResponseTo(**res)

    def update(self, id: int, dto: StickerRequestTo) -> StickerResponseTo:
        if not sticker_repo.find_by_id(id):
            raise AppException(404, "Sticker not found", 10)
        data = dto.model_dump()
        data["id"] = id
        saved = sticker_repo.save(data)
        return StickerResponseTo(**saved)

    def delete(self, id: int):
        if not sticker_repo.delete(id):
            raise AppException(404, "Sticker not found", 11)
